All_years/04-Social_Care/04-SDS_data.R

```diff
@@ -0,0 +1,152 @@
+#####################################################
+# Social Care SDS Data
+# Author: Jennifer Thom
+# Date: July 2022
+# Written on RStudio Server
+# Version of R - 3.6.1
+# Input - Data from Social care database DVPROD
+# Description - Process SDS data
+#####################################################
+
+# Load packages
+
+library(readr)
+library(dplyr)
+library(phsmethods)
+library(lubridate)
+
+
+# Set up------------------------------------------------------------------
+
+source("All_years/04-Social_Care/00-Social_Care_functions.R")
+
+
+# Read Demographic file----------------------------------------------------
+
+sc_demographics <- haven::read_sav(fs::path(
+  social_care_dir,
+  paste0("sc_demographics_lookup_", latest_update()),
+  ext = "zsav"
+))
+
+
+# Query to database -------------------------------------------------------
+
+# set-up conection to platform
+db_connection <- phs_db_connection(dsn = "DVPROD")
+
+# read in data - social care 2 demographic
+sds_full_data <- tbl(db_connection, dbplyr::in_schema("social_care_2", "sds_snapshot")) %>%
+  select(
+    sending_location,
+    social_care_id,
+    period,
+    sds_start_date,
+    sds_end_date,
+    sds_option_1,
+    sds_option_2,
+    sds_option_3
+  ) %>%
+  collect()
+
+
+# Data Cleaning-----------------------------------------------------
+
+sds_full_clean <- sds_full_data %>%
+  # Deal with SDS option 4
+  mutate(across(starts_with("sds_option_"), ~ .x == "1")) %>%
+  # SDS option 4 is derived when a person recieves more than one option.
+  # e.g. if person has option 1 and 2 then option 4 will be derived
+  mutate(sds_option_4 = rowSums(across(starts_with("sds_option_"))) > 1, .after = sds_option_3) %>%
+  # Match on demographics data (chi, gender, dob and postcode)
+  left_join(sc_demographics, by = c("sending_location", "social_care_id")) %>%
+  # If sds start date is missing, assign start of FY
+  mutate(sds_start_date = if_else(is.na(sds_start_date),
+    start_fy(year = period, format = "alternate"),
+    sds_start_date
+  )) %>%
+  # rename for matching source variables
+  rename(
+    record_keydate1 = sds_start_date,
+    record_keydate2 = sds_end_date
+  ) %>%
+  # Pivot longer on sds option variables
+  tidyr::pivot_longer(
+    cols = contains("sds_option_"),
+    names_to = "sds_option",
+    names_prefix = "sds_option_",
+    names_transform = list(sds_option = ~ paste0("SDS-", .x)),
+    values_to = "received",
+    values_transform = list(received = as.integer)
+  ) %>%
+  # Only keep rows where they received a package and remove duplicates
+  filter(received == 1) %>%
```

Comment:
We've changed them to logical above, so we can simplify this a little bit.
```suggestion
  tidyr::pivot_longer(
    cols = contains("sds_option_"),
    names_to = "sds_option",
    names_prefix = "sds_option_",
    names_transform = list(sds_option = ~ paste0("SDS-", .x)),
    values_to = "received"
  ) %>%
  # Only keep rows where they received a package and remove duplicates
  filter(received) %>%
```